### PR TITLE
512 --> 513 [WIP]

### PR DESCRIPTION
**Pull request by qwertyquerty**
## About The Pull Request

This pull request includes all the updates needed to get us onto 513 as quick as possible.

## Why It's Good For The Game

We would like to be the first server running 513

## Changelog
:cl:
tweak: Misc internal changes for 513
/:cl:

## TODO:

- [x] Guns currently seem to fire their bullets at their normal angle but reflected across the y=x axis. I believe this has something to do with the new ATAN2.

### CODE REVIEW

**Comment by qwertyquerty** [comment]
It appears the issue with the gunfire is that lummox didn't know the args were y,x and not x,y in the C atan2. It will be fixed next update.

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
-#define islist(L) (istype(L, /list))
-
 #define in_range(source, user) (get_dist(source, user) <= 1 && (get_step(source, 0)?:z) == (get_step(user, 0)?:z))
 
-#define ismovableatom(A) (istype(A, /atom/movable))
-
 #define isatom(A) (isloc(A))
 
+#define ismovableatom(A) (ismovable(A)) // we got ismovable in 513
```
Comment (ike709) [inline code/__DEFINES/is_helpers.dm:7]: Why not just do a find & replace?

File: code/__HELPERS/_lists.dm [hunk 2]
```diff
@@ -537,7 +537,7 @@
 		used_key_list[input_key] = 1
 	return input_key
 
-#if DM_VERSION > 512
+#if DM_VERSION > 513
 #error Remie said that lummox was adding a way to get a lists
 #error contents via list.values, if that is true remove this
 #error otherwise, update the version and bug lummox
```
Comment (ike709) [inline code/__HELPERS/_lists.dm:543]: Did you bug lummox?

**Comment by ike709** [comment]
Why not just find & replace all of the define uses with the new internal procs?

File: code/__HELPERS/_lists.dm (hunk 2 above)
Comment (qwertyquerty) [reply code/__HELPERS/_lists.dm:543]: Yes.

**Comment by qwertyquerty** [comment]
Increases our diff

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (qwertyquerty) [reply code/__DEFINES/is_helpers.dm:7]: increasing our diff. we can do this if TG also does it.

**Comment by qwertyquerty** [comment]
Doesn't appear to be using 513

**Comment by ike709** [comment]
Dreamchecker doesn't support 513 yet.

**Comment by Crossedfall** [comment]
Is this going to be on-hold until dreamchecker is updated or is the plan to add backwards compatibility?

**Comment by ike709** [comment]
I'd prefer the latter.

**Comment by ike709** [comment]
Still no ETA for 513 DreamChecker as far as I know.

**Comment by optimumtact** [comment]
Dreamchecker won't go 513 until tg goes 513 probably

**Pull request closed without merging**